db/migrate/20180812142030_add_group_min_trust_level.rb

```diff
@@ -0,0 +1,6 @@
+class AddGroupMinTrustLevel < ActiveRecord::Migration[5.2]
+  def change
+    add_column :users, :group_min_trust_level, :integer
+    execute "UPDATE users SET group_min_trust_level = group_locked_trust_level"
```

Comment:
This can still result in data loss because this migration will run first before the [guards](https://github.com/discourse/discourse/pull/6279/files#diff-4676c008b11a5480d73d4a6de01e45b9R310) are deployed. Note that we don't bring down all of our app servers at the same time during our deployment process. Instead it is done on a rotation basis where we bring down each app server and deploy it again with the latest changes.

Reply:
The solution we have now is the mark the old column as read-only using https://github.com/discourse/discourse/blob/master/lib/migration/column_dropper.rb#L14 but this is problematic for backups and restores. I'm trying to get it fixed by this week so maybe we can delay this till then?

Reply:
The deployment of the app servers can potentially fail after the migration has been run which may lead to a larger data loss window.